specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json

```diff
@@ -928,6 +1007,218 @@
+      "description": "The Private Endpoint network resource that is linked to the Private Endpoint connection.",
+      "readOnly": true,
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Gets or sets id.",
+          "readOnly": true,
+          "type": "string"
+        }
+      }
+    },
+    "PrivateLinkServiceConnectionState": {
+      "description": "Gets or sets private link service connection state.",
+      "readOnly": true,
+      "type": "object",
+      "properties": {
+        "status": {
+          "description": "Gets or sets the status.",
+          "readOnly": true,
+          "enum": [
+            "Pending",
+            "Approved",
+            "Rejected",
+            "Disconnected"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PrivateEndpointConnectionStatus",
+            "modelAsString": true
+          }
+        },
+        "description": {
+          "description": "Gets or sets description.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "actionsRequired": {
+          "description": "Gets or sets actions required.",
+          "readOnly": true,
+          "type": "string"
+        }
+      }
+    },
+    "PrivateLinkResources": {
+      "description": "Class which represent the stamps associated with the vault.",
+      "readOnly": true,
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkResource"
+          }
+        },
+        "nextLink": {
+          "description": "Link to the next chunk of the response",
+          "type": "string"
+        }
+      }
+    },
+    "PrivateLinkResource": {
+      "description": "Private Link Resource.",
+      "readOnly": true,
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "e.g. subscriptions/{subId}/resourceGroups/{groupName}/providers/Microsoft.RecoveryServices/vaults/{vaultName}/privateLinkResources/backupresource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "e.g. backupresource or siterecoveryresource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "e.g. Microsoft.RecoveryServices/vaults/privateLinkResources",
+          "readOnly": true,
+          "type": "string"
+        },
+        "groupId": {
+          "description": "e.g. f9ad6492-33d4-4690-9999-6bfd52a0d081 (Backup) or f9ad6492-33d4-4690-9999-6bfd52a0d082 (SiteRecovery)",
+          "readOnly": true,
+          "type": "string"
+        },
+        "requiredMembers": {
+          "description": "[backup-ecs1, backup-prot1, backup-prot1b, backup-prot1c, backup-id1]",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
```

Comment:
everything but "id", "name", and "type" should be inside a "properties" object.

Reply:
This is actually not an arm resource proxy/tracked. Network team expects these to be in this format for RPs onboarding to Private Endpoint. For eg: See already published documentation for other onboarded RP:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.eventgrid.models.privatelinkresource?view=azure-dotnet-preview

Reply:
You are linking to the .NET class, not the API.  They are using x-ms-client-flatten to flatten the resulting generated SDK.  The API does have it in a property bag.  See line 7309 in specification\eventgrid\resource-manager\Microsoft.EventGrid\preview\2020-04-01-preview\EventGrid.json.  This is the billionth review of these APIs, the contract is well known

Reply:
Thanks Chris for the explanation. We have fixed the PrivateLinkResources API response as you suggested.